app/controllers/concerns/recaptcha_checking.rb

```diff
@@ -1,24 +1,33 @@
 module RecaptchaChecking
   extend ActiveSupport::Concern
 
-  SUSPICIOUS_RECAPTCHA_THRESHOLD = 0.5
+  SUSPICIOUS_RECAPTCHA_V3_THRESHOLD = 0.5
 
-  # https://github.com/ambethia/recaptcha#verify_recaptcha
-  # https://github.com/ambethia/recaptcha#recaptcha_reply
-  def recaptcha_is_invalid?(model = nil)
-    !verify_recaptcha(model: model, action: controller_name, minimum_score: SUSPICIOUS_RECAPTCHA_THRESHOLD) && recaptcha_reply
+  # https://github.com/ambethia/recaptcha
+  # Our approach combines Recaptcha V3 with Recaptcha V2 as a fallback, as documented here:
+  # https://github.com/ambethia/recaptcha?tab=readme-ov-file#examples
+
+  def recaptcha_protected(form: nil, fallback_template: :new)
+    if recaptcha_is_valid?
+      yield
+    else
+      @show_recaptcha = true
+      form&.errors&.add(:recaptcha, t("recaptcha.error"))
+      render fallback_template
+    end
   end
 
-  def handle_invalid_recaptcha(form: nil, score: nil)
-    log_invalid_recaptcha(form: form, score: score)
-    redirect_to invalid_recaptcha_path
+  def recaptcha_is_valid?
+    recaptcha_v3_is_valid? || recaptcha_v2_is_valid?
   end
 
-  def log_invalid_recaptcha(form: nil, score: nil)
-    form_name = form.class.name.gsub("::", "").underscore.humanize if form.present?
-    Sentry.with_scope do |scope|
-      scope.set_tags("form.name": form_name, "recaptcha.score": score)
-      Sentry.capture_message("Invalid recaptcha", level: :warning)
-    end
```

Comment:
Now we fallback into a puzzle instead of recording/sending the failed score to Sentry.